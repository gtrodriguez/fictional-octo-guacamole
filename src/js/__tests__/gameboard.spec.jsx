import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import Header from '../components/header';
import GameBoard from '../components/gameboard';

describe('Gameboard', () =>{
  Enzyme.configure({ adapter: new Adapter() })

    //TEST USER
  let user = {
    username: "Player1",
    email: "Player1Email"
  },
  partner = {
    username: "Player2",
    email: "Player2Email"
  },
  connection = {},
  gameMatrix = new Array(8);

  for (var i = 0; i < 8; i += 1) {
    gameMatrix[i] = new Array(8);
    gameMatrix[i].fill(0);
  };

  let gameInstance = {
    scoreBoard: gameMatrix,
    player1: user.username,
    player2: partner.username,
    currentPlayer: user.username,
    isActive: false,
    _id: '123456',
    gameOver: false
  },
  newGameInstance = {};

  var updateGameInstance = (game) => {
    newGameInstance = Object.assign({}, game);
  };

  it('renders correctly',() => {
    const component = Enzyme.shallow(<GameBoard
      gameInstance={gameInstance}
      user={user}
      connection={connection} />);

    expect(component).toMatchSnapshot();
  });

  if('detects wins correctly', () => {
    const component = Enzyme.render(<GameBoard
      gameInstance={gameInstance}
      user={user}
      connection={connection} />);

    component.find('.interactive-tile:nth-child(1)').simulate('click');
  });
});
