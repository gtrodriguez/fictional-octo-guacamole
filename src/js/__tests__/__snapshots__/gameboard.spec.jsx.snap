// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gameboard renders correctly 1`] = `
<div
  id="game-container"
>
  <div
    className=""
    id="game-board-container"
  >
    <div
      id="control-panel"
    >
      <GameControlPanel
        gameInstance={
          Object {
            "_id": "123456",
            "currentPlayer": "Player1",
            "gameOver": false,
            "isActive": false,
            "player1": "Player1",
            "player2": "Player2",
            "scoreBoard": Array [
              Array [
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
              ],
              Array [
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
              ],
              Array [
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
              ],
              Array [
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
              ],
              Array [
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
              ],
              Array [
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
              ],
              Array [
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
              ],
              Array [
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
              ],
            ],
          }
        }
        handleGameInvite={[Function]}
        user={
          Object {
            "email": "Player1Email",
            "username": "Player1",
          }
        }
      />
    </div>
    <div>
      <div
        id="interactive-row"
      >
        <InteractiveTile
          enabled={[Function]}
          handleClick={[Function]}
          key="0"
          x={0}
        />
        <InteractiveTile
          enabled={[Function]}
          handleClick={[Function]}
          key="1"
          x={1}
        />
        <InteractiveTile
          enabled={[Function]}
          handleClick={[Function]}
          key="2"
          x={2}
        />
        <InteractiveTile
          enabled={[Function]}
          handleClick={[Function]}
          key="3"
          x={3}
        />
        <InteractiveTile
          enabled={[Function]}
          handleClick={[Function]}
          key="4"
          x={4}
        />
        <InteractiveTile
          enabled={[Function]}
          handleClick={[Function]}
          key="5"
          x={5}
        />
        <InteractiveTile
          enabled={[Function]}
          handleClick={[Function]}
          key="6"
          x={6}
        />
        <InteractiveTile
          enabled={[Function]}
          handleClick={[Function]}
          key="7"
          x={7}
        />
      </div>
    </div>
    <div>
      <div
        id="game-board"
      >
        <div
          className="game-row"
          key="0"
        >
          <GameBoardTile
            data-x={0}
            data-y={0}
            key="0"
            value=""
          />
          <GameBoardTile
            data-x={0}
            data-y={1}
            key="1"
            value=""
          />
          <GameBoardTile
            data-x={0}
            data-y={2}
            key="2"
            value=""
          />
          <GameBoardTile
            data-x={0}
            data-y={3}
            key="3"
            value=""
          />
          <GameBoardTile
            data-x={0}
            data-y={4}
            key="4"
            value=""
          />
          <GameBoardTile
            data-x={0}
            data-y={5}
            key="5"
            value=""
          />
          <GameBoardTile
            data-x={0}
            data-y={6}
            key="6"
            value=""
          />
          <GameBoardTile
            data-x={0}
            data-y={7}
            key="7"
            value=""
          />
        </div>
        <div
          className="game-row"
          key="1"
        >
          <GameBoardTile
            data-x={1}
            data-y={0}
            key="0"
            value=""
          />
          <GameBoardTile
            data-x={1}
            data-y={1}
            key="1"
            value=""
          />
          <GameBoardTile
            data-x={1}
            data-y={2}
            key="2"
            value=""
          />
          <GameBoardTile
            data-x={1}
            data-y={3}
            key="3"
            value=""
          />
          <GameBoardTile
            data-x={1}
            data-y={4}
            key="4"
            value=""
          />
          <GameBoardTile
            data-x={1}
            data-y={5}
            key="5"
            value=""
          />
          <GameBoardTile
            data-x={1}
            data-y={6}
            key="6"
            value=""
          />
          <GameBoardTile
            data-x={1}
            data-y={7}
            key="7"
            value=""
          />
        </div>
        <div
          className="game-row"
          key="2"
        >
          <GameBoardTile
            data-x={2}
            data-y={0}
            key="0"
            value=""
          />
          <GameBoardTile
            data-x={2}
            data-y={1}
            key="1"
            value=""
          />
          <GameBoardTile
            data-x={2}
            data-y={2}
            key="2"
            value=""
          />
          <GameBoardTile
            data-x={2}
            data-y={3}
            key="3"
            value=""
          />
          <GameBoardTile
            data-x={2}
            data-y={4}
            key="4"
            value=""
          />
          <GameBoardTile
            data-x={2}
            data-y={5}
            key="5"
            value=""
          />
          <GameBoardTile
            data-x={2}
            data-y={6}
            key="6"
            value=""
          />
          <GameBoardTile
            data-x={2}
            data-y={7}
            key="7"
            value=""
          />
        </div>
        <div
          className="game-row"
          key="3"
        >
          <GameBoardTile
            data-x={3}
            data-y={0}
            key="0"
            value=""
          />
          <GameBoardTile
            data-x={3}
            data-y={1}
            key="1"
            value=""
          />
          <GameBoardTile
            data-x={3}
            data-y={2}
            key="2"
            value=""
          />
          <GameBoardTile
            data-x={3}
            data-y={3}
            key="3"
            value=""
          />
          <GameBoardTile
            data-x={3}
            data-y={4}
            key="4"
            value=""
          />
          <GameBoardTile
            data-x={3}
            data-y={5}
            key="5"
            value=""
          />
          <GameBoardTile
            data-x={3}
            data-y={6}
            key="6"
            value=""
          />
          <GameBoardTile
            data-x={3}
            data-y={7}
            key="7"
            value=""
          />
        </div>
        <div
          className="game-row"
          key="4"
        >
          <GameBoardTile
            data-x={4}
            data-y={0}
            key="0"
            value=""
          />
          <GameBoardTile
            data-x={4}
            data-y={1}
            key="1"
            value=""
          />
          <GameBoardTile
            data-x={4}
            data-y={2}
            key="2"
            value=""
          />
          <GameBoardTile
            data-x={4}
            data-y={3}
            key="3"
            value=""
          />
          <GameBoardTile
            data-x={4}
            data-y={4}
            key="4"
            value=""
          />
          <GameBoardTile
            data-x={4}
            data-y={5}
            key="5"
            value=""
          />
          <GameBoardTile
            data-x={4}
            data-y={6}
            key="6"
            value=""
          />
          <GameBoardTile
            data-x={4}
            data-y={7}
            key="7"
            value=""
          />
        </div>
        <div
          className="game-row"
          key="5"
        >
          <GameBoardTile
            data-x={5}
            data-y={0}
            key="0"
            value=""
          />
          <GameBoardTile
            data-x={5}
            data-y={1}
            key="1"
            value=""
          />
          <GameBoardTile
            data-x={5}
            data-y={2}
            key="2"
            value=""
          />
          <GameBoardTile
            data-x={5}
            data-y={3}
            key="3"
            value=""
          />
          <GameBoardTile
            data-x={5}
            data-y={4}
            key="4"
            value=""
          />
          <GameBoardTile
            data-x={5}
            data-y={5}
            key="5"
            value=""
          />
          <GameBoardTile
            data-x={5}
            data-y={6}
            key="6"
            value=""
          />
          <GameBoardTile
            data-x={5}
            data-y={7}
            key="7"
            value=""
          />
        </div>
        <div
          className="game-row"
          key="6"
        >
          <GameBoardTile
            data-x={6}
            data-y={0}
            key="0"
            value=""
          />
          <GameBoardTile
            data-x={6}
            data-y={1}
            key="1"
            value=""
          />
          <GameBoardTile
            data-x={6}
            data-y={2}
            key="2"
            value=""
          />
          <GameBoardTile
            data-x={6}
            data-y={3}
            key="3"
            value=""
          />
          <GameBoardTile
            data-x={6}
            data-y={4}
            key="4"
            value=""
          />
          <GameBoardTile
            data-x={6}
            data-y={5}
            key="5"
            value=""
          />
          <GameBoardTile
            data-x={6}
            data-y={6}
            key="6"
            value=""
          />
          <GameBoardTile
            data-x={6}
            data-y={7}
            key="7"
            value=""
          />
        </div>
        <div
          className="game-row"
          key="7"
        >
          <GameBoardTile
            data-x={7}
            data-y={0}
            key="0"
            value=""
          />
          <GameBoardTile
            data-x={7}
            data-y={1}
            key="1"
            value=""
          />
          <GameBoardTile
            data-x={7}
            data-y={2}
            key="2"
            value=""
          />
          <GameBoardTile
            data-x={7}
            data-y={3}
            key="3"
            value=""
          />
          <GameBoardTile
            data-x={7}
            data-y={4}
            key="4"
            value=""
          />
          <GameBoardTile
            data-x={7}
            data-y={5}
            key="5"
            value=""
          />
          <GameBoardTile
            data-x={7}
            data-y={6}
            key="6"
            value=""
          />
          <GameBoardTile
            data-x={7}
            data-y={7}
            key="7"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
